### ISSUE TITLE
Problem with type conversion

### ISSUE THREAD

**Issue by Malve22** [OmarMuscatello/SimplePatch#1]
I found problem with this code below we get "Invalid cast from 'System.String' to 'System.Guid'". I noticed that function Convert.ChangeType is used. Here is topic on stackoverflow with problems with this function. https://stackoverflow.com/questions/3847446/extending-convert-changetype-to-produce-user-defined-types-on-request
Could you pls provide a way to insert some custom conversion mechanism. Because I think a lot requests comes with json where types are treated as string especially Guid type. It would be convenient to handle at least string => Guid conversion. And the best would be to allow injecting some custom conversion for specific types. 

```
 public class SampleClass
    {
        public Guid Id { get; set; }
        public Guid AssigneeId { get; set; }
    }
    class Program
    {
        static void Main(string[] args)
        {
            var currentTask = new SampleClass
            {
                AssigneeId = Guid.NewGuid(),
                Id = Guid.NewGuid()
            };
            Delta<SampleClass> task = new Delta<SampleClass>();
            task.Add("AssigneeId", Guid.NewGuid().ToString());

            task.Patch(currentTask);

        }
    }

```

**Comment by OmarMuscatello** [comment]
Thank you @Malve22 for interesting in [SimplePatch](https://github.com/OmarMuscatello/SimplePatch). I just published the 1.2.1 version which handles Guid type correctly. I'm working on providing a way to use custom conversion mechanism.

**Comment by Malve22** [comment]
Yes it helped thanks.
